lib/game.rb

```diff
@@ -3,25 +3,53 @@
 require_relative '../dependencies'
 
 class Game
-  include GameStart
+  # include GameStart
   include Validation
 
+  NOT_YET = '-'
+  GOT_IT = '+'
+  DIFFICULTY_LEVEL = {
+      easy: { attempts: 15, hints: 3 },
+      medium: { attempts: 10, hints: 2 },
+      hell: { attempts: 5, hints: 1 }
+  }.freeze
+
   attr_accessor :attempts_total, :attempts, :difficulty, :hints_total, :hints, :name, :win, :secret
 
   def initialize(name:, difficulty:)
     @name = name
     @difficulty = difficulty
-    @attempts = GameStart::DIFFICULTY_LEVEL[difficulty][:attempts]
-    @hints = GameStart::DIFFICULTY_LEVEL[difficulty][:hints]
+    @attempts = DIFFICULTY_LEVEL[difficulty][:attempts]
+    @hints = DIFFICULTY_LEVEL[difficulty][:hints]
     @win = false
     @secret = make_number
     @unused_hints = @secret.chars
   end
 
+
+  def make_number(chars = 4, numbers = 6)
+    (1..chars).map { rand(1..numbers) }.join
+  end
+
+  def check_numbers(secret, numbers)
+    minuses = (secret & numbers).map { |element| [secret.count(element), numbers.count(element)].min }.sum
+    result = NOT_YET * minuses
+
+    numbers.each.with_index do |number, index|
+      result.sub!(NOT_YET, GOT_IT) if number == secret[index]
+    end
+
+    result
+  end
+
+  def hint(secret)
+    secret.shuffle!.pop
+  end
+
   def check(number)
     @attempts -= 1
     result = check_numbers(@secret.chars, number.chars)
-    @win = true if result == GameStart::GOT_IT * Validation::NUMBERS
+    @win = true if result == GOT_IT * Validation::NUMBERS
```

Comment:
let's improve the naming:
Validation::NUMBERS -> Validation::SECRET_CODE_LENGTH or smth like this